### PR TITLE
Update UG and DG

### PR DESCRIPTION
**Pull request by AriellaCallista**
Changes to UG:
- Reformatted `search` and `create`

Changes to DG:
- Added "Planned Enhancements" section
- Updated `model` and `storage` class diagrams

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -617,67 +622,54 @@ In essence, this allows you to find job applicants whose performance rating is a
...
-Example:
-* `search st/interviewed`
-
-#### Searching job applicants by tag
-
-Finds job applicants whose tag(s) match any of the given tag keywords
+Format: `search (n/NAME [MORE NAME] / st/STATUS [MORE STATUS] / t/TAG [MORE TAGS)`
 
-Format: `search t/TAGNAME...`
+Prefix      | Constraints
```
Comment (ketweeen) [inline docs/UserGuide.md:631]: Consider adding type (mandatory/optional) column? Can consider doing a merged column filled with mandatory*

File: docs/UserGuide.md [hunk 2]
```diff
@@ -617,67 +622,54 @@ In essence, this allows you to find job applicants whose performance rating is a
...
+* Each prefix can only be used at most once.
+* Multiple search parameters for a specific category are divided by spaces (not commas!) i.e. `search st/preliminary interviewed`
 
-#### Notes for advanced users:
-* You can combine the search categories (e.g. `search n/Alex st/offered t/software engineer`) in a single search command.
-* Each search category can be used at most once in a single search command
-  e.g. `search n/Alex n/Adam st/rejected` is not allowed.
+<box type="tip" seamless>
+Tip:
+* You can combine multiple search categories in a single `search` command.
```
Comment (ketweeen) [inline docs/UserGuide.md:646]: Incorrect formatting

File: docs/UserGuide.md [hunk 3]
```diff
@@ -617,67 +622,54 @@ In essence, this allows you to find job applicants whose performance rating is a
...
 
-Example:
-* `search n/Alex Bernice st/interviewed rejected t/intern` will output applicants whose:
-    * names contain either Alex `or` Bernice
-    * `and` status is either interviewed `or` rejected.
-    * `and` has a tag `intern`
+Examples of successful command execution:
+1. `search n/alex bernice`
+<br>
+![search-success-1](images/search-success-1.png)
```
Comment (ketweeen) [inline docs/UserGuide.md:653]: Not sure why but picture doesn't show up

File: docs/UserGuide.md [hunk 4]
```diff
@@ -617,67 +622,54 @@ In essence, this allows you to find job applicants whose performance rating is a
...
+![search-success](images/search-success.png)
+Notice how the above `search` command did not display "Alex" despite his profile matching
+the `name` and `tag` categories. This is because `search` does an `AND` search across multiple categories.
 
-![Search](images/search-2.png)
+<box type="tip" seamless>
+What does it mean to do an `OR` search within a single category and an `AND` search across multiple categories?
+It's best to explain this by breaking down an example `search` command!
+`search n/alex bernice st/interviewed t/intern` will output applicants whose:
+  * names contain either Alex `OR` Bernice
```
Comment (ketweeen) [inline docs/UserGuide.md:676]: Incorrect formatting

**Review by ketweeen — COMMENTED** [review]
Please fix formatting issues

File: docs/UserGuide.md (hunk 1 above)
Comment (AriellaCallista) [reply docs/UserGuide.md:631]: Regarding this part, technically all fields are optional, but they need at least 1 of the "optional" fields. But I wasn't sure how to word this without making it even more confusing (as flagged by Jun Hiang during our UG consult) so I just got rid of the mandatory/optional column. Any suggestions on how to solve this?

File: docs/UserGuide.md (hunk 2 above)
Comment (AriellaCallista) [reply docs/UserGuide.md:646]: Noted!

File: docs/UserGuide.md (hunk 1 above)
Comment (ketweeen) [reply docs/UserGuide.md:631]: I simply added Mandatory* and added a clarifying note below it for delete. You can refer to image below.

image

**Review by ketweeen — APPROVED** [review]
LGTM

**Pull request merged**